pkg/start/start.go

```diff
@@ -153,13 +153,14 @@ func (o *Options) run(ctx context.Context, controllerCtx *Context, lock *resourc
 
 	// TODO: when we switch to graceful lock shutdown, this can be
 	// moved back inside RunOrDie
-	go leaderelection.RunOrDie(leaderelection.LeaderElectionConfig{
+	// TODO: properly wire ctx here
+	go leaderelection.RunOrDie(context.TODO(), leaderelection.LeaderElectionConfig{
```

Comment:
Was this not working when you passed ctx to RunOrDie?

Reply:
we have some dirty exit channel logic. Prefered to keep that "rewrite" out of the compile fixes.